src/resolution_functions/models/pychop.py

```diff
@@ -549,19 +554,60 @@ def _get_chop_times(cls,
         return chop_times
 
     @classmethod
-    def get_chopper_width_squared(cls,
-                                  model_data: PyChopModelDataNonFermi,
-                                  e_init: float,
-                                  chopper_frequency: list[int]) -> tuple[float, float]:
+    def _get_chopper_width_squared(cls,
+                                   model_data: PyChopModelDataNonFermi,
+                                   e_init: float,
+                                   chopper_frequency: list[int]) -> tuple[float, float]:
         chop_times = cls._get_chop_times(model_data, e_init, chopper_frequency)
-        chop_times = [chop_times[0][0], chop_times[-1][0]]
 
-        wd0 = (chop_times[1][1] - chop_times[1][0]) / 2.0 / 1.0e6
-        wd1 = (chop_times[0][1] - chop_times[0][0]) / 2.0 / 1.0e6
+        wd0 = (chop_times[-1][0][1] - chop_times[-1][0][0]) * 0.5e-6
+        wd1 = (chop_times[0][0][1] - chop_times[0][0][0]) * 0.5e-6
 
         return wd0 ** 2, wd1 ** 2
 
 
+class PyChopModelCNCS(PyChopModelNonFermi):
+    def __init__(self,
+                 model_data: PyChopModelDataNonFermi,
+                 e_init: Optional[float] = None,
+                 resolution_disk_frequency: Optional[int] = None,
+                 fermi_frequency: Optional[int] = None,
+                 fitting_order: Optional[int] = 4,
```

Comment:
Is there any scenario it would be `None`? If not, it isn't an `Optional` type. (Some projects and linters are replacing all Optional with `| None` to avoid confusion.)